packages/core/src/runtime.ts

```diff
@@ -352,23 +352,15 @@ export class AgentRuntime implements IAgentRuntime {
     // First create the agent entity directly
     try {
       // Ensure agent exists first (this is critical for test mode)
-      const agentExists = await this.adapter.ensureAgentExists(this.character as Partial<Agent>);
-
-      // Verify agent exists before proceeding
-      const agent = await this.adapter.getAgent(this.agentId);
-      if (!agent) {
-        throw new Error(
-          `Agent ${this.agentId} does not exist in database after ensureAgentExists call`
-        );
-      }
+      const existingAgent = await this.adapter.ensureAgentExists(this.character as Partial<Agent>);
 
       // No need to transform agent's own ID
-      const agentEntity = await this.adapter.getEntityById(this.agentId);
+      let agentEntity = await this.adapter.getEntityById(this.agentId);
 
       if (!agentEntity) {
         const created = await this.createEntity({
           id: this.agentId,
-          agentId: this.agentId,
+          agentId: existingAgent.id,
```

Comment:
Hey! I noticed one thing that might be worth double-checking:

In this block, we're using existingAgent.id when creating the entity

But we're calling getEntityById(this.agentId) both before and after that. If existingAgent.id is different from this.agentId, we might end up creating an entity with one agentId and then trying to look it up using a different one